graylog2-server/src/main/java/org/graylog2/lookup/db/DBLookupTableService.java

```diff
@@ -109,17 +114,12 @@ public Collection<LookupTableDto> findByDataAdapterIds(Collection<String> dataAd
         }
     }
 
-    private ImmutableList<LookupTableDto> asImmutableList(Iterator<? extends LookupTableDto> cursor) {
-        return ImmutableList.copyOf(cursor);
-    }
-
-    public void delete(String idOrName) {
+    @Override
+    public int delete(String idOrName) {
         final Optional<LookupTableDto> lookupTableDto = get(idOrName);
-        lookupTableDto
-                .map(LookupTableDto::id)
-                .map(ObjectId::new)
-                .ifPresent(db::removeById);
+        int numDeleted = super.delete(idOrName);
```

Comment:
While this is old code, relying on an exception as test (ID or name) is not a good approach.  In this case, we can use ObjectId to test whether idOrName is a valid ObjectId, if so, search by ID, else by name.